src/build.py

```diff
@@ -1267,47 +1266,31 @@ def LibCXXABI():
 
 def Musl():
   buildbot.Step('musl')
+
+  # Strictly speaking the CMake toolchain file isn't part of musl, but does
+  # go along with the headers and libs musl installs. Give it a special
+  # path to avoid warnings about the system being unknown.
+  Mkdir(os.path.dirname(CMAKE_TOOLCHAIN_FILE))
+  shutil.copy2(os.path.join(SCRIPT_DIR, 'wasm_standalone.cmake'),
+               CMAKE_TOOLCHAIN_FILE)
+
   Mkdir(MUSL_OUT_DIR)
-  try:
-    cc_env = BuildEnv(MUSL_OUT_DIR, use_gnuwin32=True)
-    # Build musl directly to wasm object files in an ar library
-    proc.check_call([
-        os.path.join(MUSL_SRC_DIR, 'libc.py'),
-        '--clang_dir', INSTALL_BIN,
-        '--binaryen_dir', os.path.join(INSTALL_BIN),
-        '--sexpr_wasm', os.path.join(INSTALL_BIN, 'wat2wasm'),
-        '--out', os.path.join(MUSL_OUT_DIR, 'libc.a'),
-        '--musl', MUSL_SRC_DIR, '--compile-to-wasm'], env=cc_env)
-    AR = os.path.join(INSTALL_BIN, 'llvm-ar')
-    proc.check_call([AR, 'rc', os.path.join(MUSL_OUT_DIR, 'libm.a')])
-    CopyLibraryToSysroot(os.path.join(MUSL_OUT_DIR, 'libc.a'))
-    CopyLibraryToSysroot(os.path.join(MUSL_OUT_DIR, 'libm.a'))
-    CopyLibraryToSysroot(os.path.join(MUSL_OUT_DIR, 'crt1.o'))
-    CopyLibraryToSysroot(os.path.join(MUSL_SRC_DIR, 'arch', 'wasm32',
-                                      'libc.imports'))
-
-    wasm_js = os.path.join(MUSL_SRC_DIR, 'arch', 'wasm32', 'wasm.js')
-    CopyLibraryToArchive(wasm_js)
-
-    CopyTree(os.path.join(MUSL_SRC_DIR, 'include'),
-             os.path.join(INSTALL_SYSROOT, 'include'))
-    CopyTree(os.path.join(MUSL_SRC_DIR, 'arch', 'generic', 'bits'),
-             os.path.join(INSTALL_SYSROOT, 'include', 'bits'))
-    CopyTree(os.path.join(MUSL_SRC_DIR, 'arch', 'wasm32', 'bits'),
-             os.path.join(INSTALL_SYSROOT, 'include', 'bits'))
-    CopyTree(os.path.join(MUSL_OUT_DIR, 'obj', 'include', 'bits'),
-             os.path.join(INSTALL_SYSROOT, 'include', 'bits'))
-    # Strictly speaking the CMake toolchain file isn't part of musl, but does
-    # go along with the headers and libs musl installs. Give it a special
-    # path to avoid warnings about the system being unknown.
-    cmake_path = os.path.dirname(CMAKE_TOOLCHAIN_FILE)
-    Mkdir(cmake_path)
-    shutil.copy2(os.path.join(SCRIPT_DIR, 'wasm_standalone.cmake'),
-                 CMAKE_TOOLCHAIN_FILE)
+  cc_env = BuildEnv(MUSL_OUT_DIR, use_gnuwin32=True)
 
-  except proc.CalledProcessError:
-    # Note the failure but allow the build to continue.
-    buildbot.Fail()
+  command = [PREBUILT_CMAKE_BIN, '-G', 'Ninja', MUSL_SRC_DIR,
+             '-DCMAKE_TOOLCHAIN_FILE=' + CMAKE_TOOLCHAIN_FILE]
```

Comment:
I don't think there's any way we can get around using `-DCMAKE_C_COMPILER_WORKS=ON` here, otherwise the tests will always fail to link and you'll error. at least that's what I'm seeing.

Reply:
I wonder why this is working for me ?  I'm pretty sure I did a clean build..

Reply:
So I just tried it again; If I build with no musl-out and a clean wasm-install dir, I get errors. If you just set `CMAKE_C_COMPILER_WORKS=ON`, then the build technically succeeds but all the compiler feature checks fail (due to the same link failure) which isn't ideal. I found that if I set  '-`DCMAKE_EXE_LINKER_FLAGS=-nostdlib -Wl,--allow-undefined'` then all the feature checks link successfully and seem to have the right output. Unfortunately that doesn't work when you add it to the CMakeLists.txt file, it has to be on the command line for some reason.

Reply:
(btw the --allow-undefined thing is needed because if you use `-nostdlib- to avoid linking `-lc` then you have `_start` undefined. That's kind of an ugly hack but I haven't found a better way to make the cmake checks work as intended. Also if you add the linker flags then you can remove `CMAKE_C_COMPILER_WORKS` from the musl cmake file.

Reply:
I ended up setting CMAKE_C_COMPILER_WORKS=OFF in musl's CMakeLists.txt .. which I think makes some sense.

Reply:
(I assume you meant ON...) Right, but all the feature tests still fail if you only do that.

Reply:
I mean `=ON`.  ￼See https://github.com/jfbastien/musl/pull/45